llvm/include/llvm/CodeGen/MachinePassManager.h

```diff
@@ -43,7 +41,31 @@ extern template class AnalysisManager<MachineFunction>;
 /// automatically mixes in \c PassInfoMixin.
 template <typename DerivedT>
 struct MachinePassInfoMixin : public PassInfoMixin<DerivedT> {
-  // TODO: Add MachineFunctionProperties support.
+  // MachineFunctionProperties support
+  static MachineFunctionProperties getRequiredProperties() {
+    return MachineFunctionProperties();
+  }
+
+  static MachineFunctionProperties getSetProperties() {
+    return MachineFunctionProperties();
+  }
+
+  static MachineFunctionProperties getClearedProperties() {
+    return MachineFunctionProperties();
+  }
+};
+
+/// A CRTP mix-in that provides informational APIs needed for MachineFunction
+/// analysis passes. See also \c PassInfoMixin.
+template <typename DerivedT>
+struct MachineFunctionAnalysisInfoMixin
```

Comment:
I'm not super familiar with codegen verification. I vaguely understand that there are different parts of the pipeline that require different types of verification. but do machine function analyses also need this?